src/base/http/requestparser.cpp

```diff
@@ -109,23 +110,36 @@ RequestParser::ParseResult RequestParser::doParse(const QByteArray &data)
         return {ParseStatus::OK, m_request, headerLength};
     if (m_request.method == HEADER_REQUEST_METHOD_POST)
     {
-        bool ok = false;
-        const int contentLength = m_request.headers[HEADER_CONTENT_LENGTH].toInt(&ok);
-        if (!ok || (contentLength < 0))
+        const auto parseContentLength = [this]() -> std::optional<int>
```

Comment:
IMO, it's enough to return `-1` in case of parsing error. Otherwise you should return either a valid (non-zero) value or `std::nullopt`. The current implementation looks more like `parseInt`, which in this case is too general and leads only to additional concerns from the calling side.

Reply:
OK, pushed another version.